bitnami/neo4j/templates/tls-secret.yaml

```diff
@@ -0,0 +1,99 @@
+{{- /*
+Copyright Broadcom, Inc. All Rights Reserved.
+SPDX-License-Identifier: APACHE-2.0
+*/}}
+
+{{- if .Values.ingress.enabled }}
+{{- if .Values.ingress.secrets }}
+{{- range .Values.ingress.secrets }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .name }}
+  namespace: {{ include "common.names.namespace" $ | quote }}
+  labels: {{- include "common.labels.standard" ( dict "customLabels" $.Values.commonLabels "context" $ ) | nindent 4 }}
+    app.kubernetes.io/component: neo4j
+  {{- if $.Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" $.Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+type: kubernetes.io/tls
+data:
+  tls.crt: {{ .certificate | b64enc }}
+  tls.key: {{ .key | b64enc }}
+---
+{{- end }}
+{{- end }}
+{{- if and .Values.ingress.tls .Values.ingress.selfSigned }}
+{{- $secretName := printf "%s-tls" .Values.ingress.hostname }}
+{{- $ca := genCA "neo4j-ca" 365 }}
+{{- $cert := genSignedCert .Values.ingress.hostname nil (list .Values.ingress.hostname) 365 $ca }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ $secretName }}
+  namespace: {{ include "common.names.namespace" . | quote }}
+  labels: {{- include "common.labels.standard" ( dict "customLabels" .Values.commonLabels "context" $ ) | nindent 4 }}
+    app.kubernetes.io/component: neo4j
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+type: kubernetes.io/tls
+data:
+  tls.crt: {{ include "common.secrets.lookup" (dict "secret" $secretName "key" "tls.crt" "defaultValue" $cert.Cert "context" $) }}
+  tls.key: {{ include "common.secrets.lookup" (dict "secret" $secretName "key" "tls.key" "defaultValue" $cert.Key "context" $) }}
+  ca.crt: {{ include "common.secrets.lookup" (dict "secret" $secretName "key" "ca.crt" "defaultValue" $ca.Cert "context" $) }}
+---
+{{- end }}
+{{- end }}
+{{- if and .Values.tls.https.enabled .Values.tls.https.autoGenerated (not .Values.tls.https.existingSecret) (include "neo4j.host" .)}}
+{{- $secretName := printf "%s-https-certs" (include "common.names.fullname" .) }}
+{{- $ca := genCA "neo4j-https-ca" 365 }}
```

Comment:
I did it this way on purpose because it could be problematic when enabling TLS during upgrades.

For example:
- Deploy without TLS.
- Upgrade the chart enabling TLS for HTTPS server.
- Upgrade again, now enable TLS for both Bolt and HTTPS.

HTTPS will reuse the certs from the existing secret, and Bolt will need a new CA to be generated, therefore you would end up with two different certs from different CAs, but a very complex issue where apparently both Bolt and HTTPS certs are signed by the same CA but they are not.

> That scenario is closer to a production deployment and it should reduce complexity and possible problems.
Autogenerated certificates are not intended for production environments, so there is no point on making them production

Reply:
I've got your point, but I think it comes from a general problem that we are introducing, the ca certificate should have it's own TLS secret instead of including just only the public part in the other certificates, loosing its private key. Anyway that's another problem that should be discussed in other forum.

Reply:
I modified the secrets to share the same CA, although the risk of what I mentioned exists.

I also moved the Ingress secrets into a separated manifest